src/resampling.jl

```diff
@@ -129,3 +130,221 @@ Upsamples by a factor of two and applies the abs2 operation. The code is optimiz
 function upsample2_abs2(mat::AbstractArray{T, N}; dims=1:N) where {T,N}
     return abs2.(upsample2(mat, dims=dims))
 end
+
+"""
+    resample_czt(arr, rel_zoom; shear=nothing, shear_dim=nothing, fix_nyquist=false, new_size = size(arr), rel_pad=0.2)
+
+resamples the image with fixed factors or a list of separable functions using the chirp z transform algorithm.
+The data is first padded by a relative amount `rel_pad` which is needed to avoid wrap-around problems.
+As opposed to `resample()`, this routine allows for arbitrary non-integer zoom factors.
+It is reasonably fast but only allows a stretch (via `rel_zoom`) and a shift (via `shear` in pixels) per line or column
+
+normalize=true by default multiplies by an appropriate factor so that the array size is included in the scaling. 
+This results in an array having roughly the same mean intensity.
+
+
+# Examples
+```jdoctest
+julia> using TestImages, NDTools, View5D
+
+julia> a = Float32.(testimage("resolution"));
+
+julia> b = resample_czt(a, (0.99,1.3)); # just zooming and shrinking
+
+julia> c = resample_czt(a, (x->0.8+x^2/3,x->1.0+0.2*x)); # a more complicated distortion
+
+julia> d = resample_czt(a, (x->1.0,x->1.0), shear=(x->50*x^2,0.0)); # a more complicated distortion
+
+julia> @ve a,b,c,d # visualize distortions
+```
+"""
+function resample_czt(arr::AbstractArray{T,N}, rel_zoom; shear=nothing, shear_dim=nothing, fix_nyquist=false, new_size = size(arr), rel_pad=0.2, do_damp=false, center=CtrMid) where {T,N}
+    RT = real(T)
+    orig_size = size(arr)
+    if do_damp
+        arr = damp_edge_outside(arr, rel_pad)
+    else
+        arr = copy(arr)
+    end
+    for d in 1:length(rel_zoom)
+        sd = mod(d,ndims(arr))+1
+        if !isnothing(shear_dim)
+            sd = shear_dim[d]
+        end
+        my_zoom = 1.0
+        # case of a list of zoom numbers
+        if (isa(rel_zoom, Tuple) && isa(rel_zoom[1], Number)) || isa(rel_zoom, Number)
+            my_zoom = rel_zoom[d]
+            f_res = ift(arr, d)
+            myshear = 0
+            if !isnothing(shear)
+                myshear =  shear[d]
+            end
+            if myshear != 0
```

Comment:
`if !iszero(myshear)`

Reply:
still todo

Reply:
I disagree. We have to test for the user argument not something that is anyway set to zero in the line before.

Reply:
Is just good style, always use `iszero`

Reply:
`! iszero(myshear)`

Reply:
Other line. Not I get it.
Done

Reply:
Now I get it is what I meant ;-)